### PR TITLE
Fix CLANG Analyze

### PR DESCRIPTION
**Pull request by siying**
Summary: clang analyze shows warnings after we upgrade the CLANG version. Fix them.

Test Plan: Run all existing tests.

### CODE REVIEW

**Comment by yiwu-arbug** [comment]
I'm not familiar with compaction_job.cc, otherwise the patch looks good.
Leave to @ajkr to take a look at changes in compaction_job.cc.

File: db/compaction_job.cc [hunk 1]
```diff
@@ -1144,17 +1146,24 @@ Status CompactionJob::FinishCompactionOutputFile(
                      meta->marked_for_compaction ? " (need compaction)" : "");
     }
   }
-  std::string fname = TableFileName(db_options_.db_paths, meta->fd.GetNumber(),
-                                    meta->fd.GetPathId());
+  std::string fname;
+  FileDescriptor output_fd;
+  if (meta != nullptr) {
```
Comment (ajkr) [inline db/compaction_job.cc]: can you just add `assert(meta != nullptr)` to fix this? it's a pointer to a stack object so should be ok

**Comment by ajkr** [comment]
can you land?

File: db/compaction_job.cc (hunk 1 above)
Comment (siying) [reply db/compaction_job.cc]: Why does it a stack object? It is not clear to me when I look at the code. I guess it is not clear to CLANG either.

**Comment by siying** [comment]
Let me land it now. We can follow up the discussion on whether `meta` can be null.

**Pull request closed without merging**